democracy_club/apps/donations/helpers.py

```diff
@@ -18,50 +16,68 @@
 
 
 class GoCardlessHelper(object):
-    def __init__(self):
-        self.gocardless = gocardless
+    def __init__(self, request):
+        self.request = request
+        # We need the session to be saved before it gets a session_key
+        # This is because the code is called from a middleware
+        self.request.session.save()
+
         if getattr(settings, 'GOCARDLESS_USE_SANDBOX', False):
-            self.gocardless.environment = "sandbox"
-        self.gocardless.set_details(
-            app_id=settings.GOCARDLESS_APP_ID,
-            app_secret=settings.GOCARDLESS_APP_SECRET,
-            access_token=settings.GOCARDLESS_ACCESS_TOKEN,
-            merchant_id=settings.GOCARDLESS_MERCHANT_ID,
-        )
+            gc_environment = "sandbox"
+            gc_token = settings.GOCARDLESS_ACCESS_TOKEN
```

Comment:
Should this line be out-dented by one tab? It looks like if sandbox mode is off you are going to try to construct `Client()` with a `gc_token` var that hasn't been initialised?

Reply:
Yep, good catch, fixedup pushed